### PR TITLE
soc: esp32s3: fix boot sequence of appcpu when AMP mode is used, make the sequence similar as the used by the esp32.

### PR DESCRIPTION
**Pull request by uLipe**
The boot sequence from esp32s3 when AMP mode is used is broken because it needs to use the same order as did in esp32, and it also needs the trick of commanding uart0 using the rom functions to get the cpu properly started.

### CODE REVIEW

File: soc/espressif/esp32s3/esp32s3-mp.c [hunk 1]
```diff
@@ -33,9 +34,27 @@ void esp_appcpu_start(void *entry_point)
...
+	 * Note that the logging done here is ACTUALLY REQUIRED FOR RELIABLE
+	 * OPERATION!  At least one particular board will experience spurious
+	 * hangs during initialization (usually the APPCPU fails to start at
+	 * all) without these calls present.  It's not just time -- careful
+	 * use of k_busy_wait() (and even hand-crafted timer loops using the
+	 * Xtensa timer SRs directly) that duplicates the timing exactly still
+	 * sees hangs.  Something is happening inside the ROM UART code that
+	 * magically makes the startup sequence reliable.
+	 *
+	 * Leave this in place until the sequence is understood better.
```
Comment (tejlmand) [inline soc/espressif/esp32s3/esp32s3-mp.c:49]: and two years later you have still not understood the startup sequence better and know why this is needed ?
Ref: https://github.com/zephyrproject-rtos/zephyr/pull/51314

Comment (uLipe) [reply soc/espressif/esp32s3/esp32s3-mp.c:49]: Uhm yes? There is a point that the uart0 rom code does an extra work that releases the appcpu.

Unfortunately I don't work for espressif and I don't have access to the rom code or the developer team.

Comment (uLipe) [reply soc/espressif/esp32s3/esp32s3-mp.c:49]: Yes, I don't have access to that documentation and the extra work that is done by the uart0.

Comment (teburd) [reply soc/espressif/esp32s3/esp32s3-mp.c:49]: Can you provide a link to the espidf code that does this?

**Comment by uLipe** [comment]
Hmm, it seems I was tricked by the APPCPU not being enabled, the fix is not necessary, at least for S3, so I'm closing this for now.

**Pull request closed without merging**